integration/tests/cook/test_basic.py

```diff
@@ -115,6 +115,35 @@ def test_progress_update_submit(self):
             self.assertEqual('Twenty-five percent', job['instances'][0]['progress_message'], message)
             self.assertIsNotNone(job['instances'][0]['sandbox_directory'], message)
 
+    def test_configurable_progress_update_submit(self):
+        job_executor_type = util.get_job_executor_type(self.cook_url)
+        command = 'echo "message: 25 Twenty-five percent" > progress_file.txt; sleep 1; exit 0'
+        job_uuid, resp = util.submit_job(self.cook_url, command=command, executor=job_executor_type, max_runtime=60000,
+                                         progress_output_file='progress_file.txt',
+                                         progress_regex_string='message: (\d*) (.*)')
+        self.assertEqual(201, resp.status_code, msg=resp.content)
+        job = util.wait_for_job(self.cook_url, job_uuid, 'completed')
+        message = json.dumps(job, sort_keys=True)
+        self.assertEqual('progress_file.txt', job['progress_output_file'], message)
+        self.assertEqual('message: (\d*) (.*)', job['progress_regex_string'], message)
+        self.assertEqual(1, len(job['instances']))
+        message = json.dumps(job['instances'][0], sort_keys=True)
+        self.assertEqual('success', job['instances'][0]['status'], message)
+        self.assertEqual('success', job['instances'][0]['status'], message)
+
+        # allow enough time for progress updates to be submitted
+        publish_interval_ms = util.get_in(util.settings(self.cook_url), 'progress', 'publish-interval-ms')
+        wait_publish_interval_secs = min(2 * publish_interval_ms / 1000, 20)
+        time.sleep(wait_publish_interval_secs)
```

Comment:
Can we accompish this with "wait for" instead of sleeping?

Reply:
The issue is that the progress field can change after the `exit_code` or `end_time` have been provided. So we can't wait for those fields to be set or for the progress field to be set.

I can move the sleep inside a `if job_executor_type == 'cook':` if you like, that way tests not using the cook executor will not need to sleep. In any case, the sleep is capped at 20 seconds.

Reply:
Done.